File: app/src/main/java/quangnam/com/sample/ui/fragment/test/TestFragmentModule.java
package quangnam.com.sample.ui.fragment.test;

import android.support.v4.app.Fragment;

import dagger.Module;
import dagger.Provides;
import quangnam.com.sample.di.PerFragment;
import quangnam.com.sample.di.module.BaseFragmentModule;

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

@Module(includes = {BaseFragmentModule.class})
public class TestFragmentModule {

    @Provides
    @PerFragment
    Fragment bindFragment(TestFragment fragment) {
        return fragment;
    }

    @Provides
    @PerFragment
    ITestFragment.IView provideView(TestFragment fragment) {
        return fragment;
    }

    @Provides
    @PerFragment
    ITestFragment.IPresenter providePresenter(ITestFragment.IView view) {
        return new TestPresenter();
    }

}

File: app/src/main/java/quangnam/com/sample/module/test/fragment/TestFragmentModule.kt
package quangnam.com.sample.module.test.fragment

import android.support.v4.app.Fragment
import dagger.Module
import dagger.Provides
import quangnam.com.sample.di.PerChildFragment
import quangnam.com.sample.di.module.BaseChildFragmentModule

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

@Module(includes = [(BaseChildFragmentModule::class)])
class TestFragmentModule {

    @Provides
    @PerChildFragment
    fun bindFragment(fragment: TestFragment): Fragment = fragment

}
